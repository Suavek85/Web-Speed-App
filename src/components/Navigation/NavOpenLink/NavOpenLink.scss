.button__wrapper {
    padding-left: 20px;
    cursor: pointer;
}

:export {
    blockWrapper: button__wrapper;
}