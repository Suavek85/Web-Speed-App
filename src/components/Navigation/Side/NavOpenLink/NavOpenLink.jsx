import React from 'react'
import { useDispatch } from "react-redux"

import { navigationStates } from '../../../../constants/navigationStates'
import { SETTINGS } from '../../../../constants/content'
import styles from './NavOpenLink.scss'

export default function NavOpenLink() {

  const dispatch = useDispatch()
  const handleExpandMenu = () => dispatch({ type: navigationStates.EXPAND })

  return (
    <div id='buttonwrapper' className={ styles.block } onClick={ handleExpandMenu }>
      <h2> { SETTINGS } </h2>
    </div>
  )
}