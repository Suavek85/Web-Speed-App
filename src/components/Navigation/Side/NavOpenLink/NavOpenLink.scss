$block: navopen;

.#{$block} {
    padding-left: 20px;
    cursor: pointer;
}

:export {
    block: $block;
}