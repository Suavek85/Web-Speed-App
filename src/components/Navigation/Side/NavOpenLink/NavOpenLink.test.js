import * as React from 'react'
import { shallow } from 'enzyme'
import * as redux from 'react-redux'
import NavOpenLink from './NavOpenLink'

it('should match the snapshot', () => {
  const mockDispatch = jest.fn()
  const spyUseDispatch = jest.spyOn(redux, 'useDispatch')
  spyUseDispatch.mockReturnValue(mockDispatch)
  const wrapper = shallow(<NavOpenLink />)
  expect(wrapper).toMatchSnapshot()
})

it('should dispatch action on click', () => {
  const mockDispatch = jest.fn()
  const spyUseDispatch = jest.spyOn(redux, 'useDispatch')
  spyUseDispatch.mockReturnValue(mockDispatch)
  const wrapper = shallow(<NavOpenLink />)
  const button = wrapper.find('#buttonwrapper')
  button.simulate('click')
  expect(mockDispatch).toHaveBeenCalledTimes(1)
})