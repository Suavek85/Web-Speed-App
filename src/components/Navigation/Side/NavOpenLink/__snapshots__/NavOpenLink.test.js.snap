// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div
  id="buttonwrapper"
  onClick={[Function]}
>
  <h2>
     
    Settings
     
  </h2>
</div>
`;
